### ISSUE TITLE
lint/style/noNonNullAssertion should report on unnecessary non-null assertions

### ISSUE THREAD

**Issue by hornta** [biomejs/biome#2950]
Having the rule `lint/style/noNonNullAssertion` report on code that is not nullable for example this code should be reported: 

```ts
let foo = "string";
if(foo!) {}
```

### Playground Link
https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAZgBvAG8AIAA9ACAAIgBhAGIAYwAiADsACgAKAC8ALwAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAAvAHMAdAB5AGwAZQAvAG4AbwBOAG8AbgBOAHUAbABsAEEAcwBzAGUAcgB0AGkAbwBuADoAIAA8AGUAeABwAGwAYQBuAGEAdABpAG8AbgA%2BAAoAaQBmACgAZgBvAG8AIQApACAAewAKACAAIABmAG8AbwA7AAoAfQA%3D

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by hornta** [comment]
Added new playground link with an example of code that should be reported.

**Comment by Conaclos** [comment]
I don't understand, the code is null assertion reported?
